R/data-objects.R

```diff
@@ -121,29 +222,82 @@ iput_ <- function(x, offset, count, logical_path, truncate, verbose) {
   )
 
   # http call
-  out <- irods_rest_call("stream", "PUT", args, verbose, x)
-
-  # response
-  invisible(out)
+  irods_rest_call("stream", "PUT", args, verbose, local_path)
 }
+
+
 #' @rdname iput
 #'
 #' @export
 iget <- function(
     logical_path,
-    path = ".",
+    local_path = basename(logical_path),
     offset = 0,
-    count = 3000L,
+    count = 2000L,
     verbose = FALSE,
     overwrite = FALSE
   ) {
 
-  # local path
-  path <- file.path(path, basename(logical_path))
+  # expand logical path to absolute logical path
+  logical_path <- get_absolute_lpath(logical_path)
+
+  # handle iRODS to local file conversion
+  resp <- irods_to_local(
+    logical_path = logical_path,
+    offset = offset,
+    count = count,
+    verbose = verbose
+  )
+
+  # check for local file
+  stop_local_overwrite(overwrite, local_path)
+
+  # close connection on exit
+  on.exit(close(con))
+
+  # write file
+  con <- file(local_path, "wb")
+  writeBin(resp, con)
+
+  # return
+  invisible(NULL)
+}
+
+#' @rdname isaveRDS
```

Comment:
isaveRDS does not match ireadRDS just below

Reply:
In which sense? (If it's documentation, I will split them)

Reply:
yes, splitting is the answer, just noticed it didn't match.